### PR TITLE
makes status defaulting more resilient and dynamic

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: create.go [hunk 1]
```diff
@@ -263,6 +285,7 @@ func (r *Resource) computePatches(ctx context.Context, accessor metav1.Object, o
 
 	// TODO emit metrics when update did not complete within a certain timeframe
 	// TODO update status condition when guest cluster is migrating from creating to created status
+	// TODO update status condition to Deleting during delete event when guest cluster is being deleted
```
Comment (xh3b4sd) [inline create.go]: Doing this in the next PRs. Just some notes for me. No need to track issues for these IMO.

File: create.go [hunk 2]
```diff
@@ -142,17 +142,39 @@ func (r *Resource) computePatches(ctx context.Context, accessor metav1.Object, o
...
+			})
+		}
+
+		if conditionsEmpty {
+			patches = append(patches, Patch{
+				Op:    "replace",
+				Path:  "/status/cluster/conditions",
+				Value: []providerv1alpha1.StatusClusterCondition{},
+			})
+		}
```
Comment (xh3b4sd) [inline create.go]: I want the defaulting to be more reliable and dynamic so this is a bit separated now and every status property managed by the resource is managed in its own dedicated way based on its own assumptions.

**Review by rossf7 — APPROVED** [review]
Nice, LGTM.

**Comment by xh3b4sd** [comment]
Yeah it is pretty cool. Whatever happens to the status, say the `versions` or `nodes` get lost for any reason, the structures are initialized from scratch and prevents falling down completely.

**Pull request merged**